app/http.php

```diff
@@ -0,0 +1,781 @@
+
+        $log->setAction($action);
+
+        $isProduction = App::getEnv('OPENRUNTIMES_ENV', 'development') === 'production';
+        $log->setEnvironment($isProduction ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+
+        $responseCode = $logger->addLog($log);
+        Console::info('Executor log pushed with status code: ' . $responseCode);
+    }
+
+    Console::error('[Error] Type: ' . get_class($error));
+    Console::error('[Error] Message: ' . $error->getMessage());
+    Console::error('[Error] File: ' . $error->getFile());
+    Console::error('[Error] Line: ' . $error->getLine());
+};
+
+App::post('/v1/runtimes')
+    ->desc("Create a new runtime server")
+    ->param('runtimeId', '', new Text(128), 'Unique runtime ID.')
+    ->param('source', '', new Text(0), 'Path to source files.')
+    ->param('destination', '', new Text(0), 'Destination folder to store build files into.')
+    ->param('vars', '', new Assoc(), 'Environment Variables required for the build')
+    // refactor to `name`
+    ->param('runtime', '', new Text(128), 'Runtime for the cloud function')
+    ->param('baseImage', '', new Text(128), 'Base image name of the runtime')
+    ->inject('orchestrationPool')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, string $source, string $destination, array $vars, string $runtime, string $baseImage, $orchestrationPool, $activeRuntimes, Response $response) {
+
+        // TODO: Check if runtime already exists..
+        $container = 'runtime-' . $runtimeId;
+
+        if ($activeRuntimes->exists($container)) {
+            throw new Exception('Runtime already exists.', 409);
+        }
+
+        $build = [];
+        $buildId = '';
+        $buildStdout = '';
+        $buildStderr = '';
+        $buildStart = \time();
+        $buildEnd = 0;
+
+        try {
+            Console::info('Building runtime with ID : ' . $runtimeId);
+            /** 
+             * Temporary file paths in the executor 
+             */
+            $tmpSource = "/tmp/$runtimeId/code.tar.gz";
+            $tmpBuildDir = "/tmp/$runtimeId/builds";
+            $tmpBuild = "/tmp/$runtimeId/builds/code.tar.gz";
+
+            /**
+             * Copy code files from source to a temporary location on the executor
+             */
+            $device = new Local($destination);
+            $buffer = $device->read($source);
+            if(!$device->write($tmpSource, $buffer)) {
+                throw new Exception('Failed to copy source code to temporary directory', 500);
+            };
+
+            /**
+             * Create a temporary folder to store builds
+             */
+            if (!\file_exists($tmpBuildDir)) {
+                if (!@\mkdir($tmpBuildDir, 0755, true)) {
+                    throw new Exception("Can't create directory : $tmpBuildDir", 500);
+                }
+            }
+
+            /**
+             * Create container
+             */
```

Comment:
We should try and see if we can use a single `$orchestration->run()` call by passing different entrypoints from the API signature. I'll see if I have better ideas after looking at the Appwrite implementation.